packages/strapi-admin/services/user.js

```diff
@@ -89,6 +90,31 @@ const updateById = async (id, attributes) => {
   return strapi.query('user', 'admin').update({ id }, attributes);
 };
 
+/**
+ * Reset a user password by email. (Used in admin:reset CLI)
+ * @param {string} email - user email
+ * @param {string} password - new password
+ */
+const resetPasswordByEmail = async (email, password) => {
+  const user = await findOne({ email });
+
+  if (!user) {
+    throw new Error(`User not found for email: ${email}`);
+  }
+
+  try {
+    await passwordValidator.validate(password);
+  } catch (error) {
+    throw new Error(
+      'Invalid password. Expected a minimum of 8 characters with at least one number and one uppercase letter'
+    );
+  }
+
+  const { id: userId } = user;
+
+  await updateById(userId, { password });
```

Comment:
Why not that's what I would usually do. FYI code style outside of the eslint rules (that can be updated ofc) should not be reviewed to avoid endless debates.